File: 0094-binary-tree-inorder-traversal/solution.c
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     struct TreeNode *left;
 *     struct TreeNode *right;
 * };
 */
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */

void inorder(struct TreeNode* root,int* res,int* i){
    if(root == NULL){
        return;
    }
    inorder(root->left,res,i);
    res[(*i)++] = root->val;
    inorder(root->right,res,i);
}
int* inorderTraversal(struct TreeNode* root, int* returnSize) {
    int* res = (int*)malloc(sizeof(int)*100);
    int ind =0;
    inorder(root,res,&ind);
    *returnSize = ind;
    return res;
}

File: 0094-binary-tree-inorder-traversal/solution.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<Integer> inorderTraversal(TreeNode root) {
        List<Integer> res = new ArrayList<>();
        if(root == null) return res;
        Stack<TreeNode> stack = new Stack<>(); 
        
        TreeNode cur = root;

        
        while(true){
            while(cur != null){
                stack.push(cur);
                cur = cur.left;
            }
            if(stack.isEmpty()){
                break;
            }
            TreeNode temp = stack.pop();
            res.add(temp.val);
            cur = temp.right;
        }
        return res;
    }
}
